### PR TITLE
WIP: Add deployment for Prometheus operator

### PR DESCRIPTION
**Pull request by rewantsoni**
## Description

Add deployment for Prometheus operator

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Review by tchughesiv — CHANGES_REQUESTED** [review]
@rewantsoni since prometheus is an officially release operator in the operatorhub, we should instead leverage olm dependencies for this. Rather than adding the prom operator deployment objects manually, please explore adding the operator as a dependency here - 
https://github.com/RHEcosystemAppEng/dbaas-operator/blob/9f2727b99335384107da1115da7690c352da63ee/bundle/metadata/dependencies.yaml

https://docs.openshift.com/container-platform/4.6/operators/understanding/olm/olm-understanding-dependency-resolution.html#olm-bundle-format-dependencies_olm-understanding-dependency-resolution

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: config/rbac/prometheus-operator-cluster-role-binding.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: prometheus-operator
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: prometheus-operator
+subjects:
+  - kind: ServiceAccount
+    name: prometheus-operator-sa
+    namespace: default
```
[the remark this answers is not in the document]
Comment (gobindadas) [reply config/rbac/prometheus-operator-cluster-role-binding.yaml:13]: We already tried with dependencies.yaml way but the problem is dbaas-operator support "AllNamespaces" where as prometheus is "SingleNamespace". So always prometheus installation fails saying "AllNamespaces InstallModeType not supported, cannot configure to watch all namespaces".

And regarding namespace: As prometheus supports only SingleNamespace, prometheus will be monitoring "openshift-dbaas-operator" , can't we  use this to install prometheus and monitor?

**Comment by rewantsoni** [comment]
Closing this PR, a fresh PR with a working code will be opened soon.

**Comment by rewantsoni** [comment]
https://github.com/RHEcosystemAppEng/dbaas-operator/pull/151

**Pull request closed without merging**